includes/MediaWiki/Api/EditEndpoint.php

```diff
@@ -137,27 +137,40 @@ public function run() {
 		$reconcileUrlProperty = new PropertyId( $inputReconcile['urlReconcile'] );
 		// For now this property must be of URL type
 		if ( $this->propertyDataTypeLookup->getDataTypeIdForProperty( $reconcileUrlProperty ) !== 'url' ) {
-			die( 'urlReconcile property must be of type url' );
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-invalid-type-property-must-be-url' ),
```

Comment:
I think it would be useful to include the actual datatype as a message argument here.

Reply:
Agree. Done.